_posts/2023-06-11-vanilla-os-orchid-devlog.md

```diff
@@ -0,0 +1,195 @@
+
+With our transition towards image-based updates, we found the need for an installer that supports installing an image-based OS. While Distinst has served well as the backend for previous versions of [Vanilla Installer](https://github.com/Vanilla-OS/vanilla-installer), we decided that the best course of action was to work on our own installation backend, which will give us the flexibility to implement new and distinct features in the upcoming releases. We call this installation tool Albius, in reference to Edmond Albius, the horticulturalist who discovered the method to pollinate the Vanilla flowers manually.
+
+Albius is written in Go and can install virtually any Linux-based operating system with the right group of commands, including distributions that use Squashfs for installation. Albius reads a JSON file containing the installation recipe and does the rest automatically, allowing for easier integration with applications written in any programming language.
+
+One of the most significant features Albius introduces is LUKS2 encryption. LUKS2 allows you to encrypt your system for enhanced security and privacy. It was one of most highly requested features from our laptop users. Now they can carry their laptops with them without any worries.
+
+## Sudon't
+
+During the design process of Orchid, one of the discussions that frequently arose was about the usefulness of the `sudo` command. Vanilla OS is an immutable system, and any modifications should occur through Apx, VSO, or ABRoot (especially with Orchid). We questioned whether it was worth including a command like `sudo` in the distribution and what benefits its removal would bring. From our analysis and discussions we identified several advantages of removing the `sudo` command.
+
+By eliminating `sudo`, the risk of accidental or harmful operations performed by unauthorized entities would get reduced. Since Vanilla OS promotes a secure and stable environment, we are limiting privileged access to help preserve the system's integrity. The absence of `sudo` would encourage the adoption of a more structured approach to privilege management. Tools like Apx, VSO, and ABRoot offer specific mechanisms for system administration, allowing users to perform configuration and update operations securely and in a controlled manner. By removing sudo, the use of specialized tools will get promoted instead of relying on potentially misused commands. Additionally, utilities like systemd already allow for sudo-less management of units, and we can't think of any other use cases where sudo would be necessary.
+
+For this reason we have chosen to go down this path and remove the `sudo` command. However, the `pkexec` command remains available, which is more suitable for a desktop environment. All your operations requiring root privileges will function the same, with the only difference is another command to invoke the root privileges, which does that in a safer manner than `sudo`.
```

Comment:
Maybe pkexec can be explained in a bit more detail? The reader will not necessarily know what pkexec is.

Reply:
The technicalities won't really matter. As explained in the paragraph, operations will be the same, just another command to invoke the root privileges:
```sudo nano /etc/example.json```
becomes
```pkexec nano /etc/example.json```

Reply:
> The technicalities won't really matter.

The line "However, the `pkexec` command **remains available**, which is more suitable for a desktop environment." kind of allude that the reader knows what `pkexec` is. I am not suggesting going into technical details, but maybe it can be reworded to make better sense.

Something like "However you can do everything you need to do as root by using another command call `pkexec`, which is more suitable for a desktop environment."

Reply:
> > The technicalities won't really matter.
> 
> The line "However, the `pkexec` command **remains available**, which is more suitable for a desktop environment." kind of allude that the reader knows what `pkexec` is. I am not suggesting going into technical details, but maybe it can be reworded to make better sense.
> 
> Something like "However you can do everything you need to do as root by using another command called `pkexec`, which is more suitable for a desktop environment."

That's exactly what I'm trying to imply with:
> All your operations requiring root privileges will function the same, with the only difference is another command to invoke the root privileges, which does that in a safer manner than `sudo`.